src/http-adapter.js

```diff
@@ -112,36 +123,30 @@ class HttpAdapter {
     return promise
   }
 
-  _makeRequest ({ url, type, headers, data }) {
-    return new Promise((resolve, reject) => {
-      let request = {
-        url,
-        type,
-        headers,
-        success: (data, textStatus, jqXhr) => {
-          if (!data && jqXhr.status !== 204) {
-            throw new Error(`request returned ${jqXhr.status} status without data`)
-          }
-          return resolve(data)
-        },
-        error: (response) => {
-          if (response.readyState === 0 || response.status === 0) {
-            // this is a canceled request, so we literally should do nothing
-            return
-          }
-
-          const error = new Error(`request for resource, ${url}, returned ${response.status} ${response.statusText}`)
-          error.response = response
-          reject(error)
-        },
-        // being explicit about data type so jQuery doesn't "intelligent guess" wrong
-        // changing this may not break tests, but does behave badly in prod
-        dataType: 'text',
-        data
-      }
-
-      ajax(request)
-    })
+  async _fetch ({ url, init }) {
+    let response = await fetch(url, init)
+    switch (response.status) {
+      case 200:
+      case 201:
+      case 202:
+      case 203:
+      case 205:
+      case 206:
+        let data = await response.json()
+        return data
```

Comment:
Just `return await response.json()`...?

I kinda feel like we should have a flag for vetting _more_ than the JSON body back... For when we need to inspect the headers, for example.

Reply:
Ill refactor that. Yeah I agree but right now we dont have a mechanism for that in place. should we add it?

Reply:
No, neither do we have a use case, currently, for checking the headers of a response.